lightning/restore/tidb.go

```diff
@@ -40,6 +44,25 @@ type TiDBManager struct {
 	parser *parser.Parser
 }
 
+// getSQLErrCode returns error code if err is a mysql error
+func getSQLErrCode(err error) (terror.ErrCode, bool) {
+	mysqlErr, ok := errors.Cause(err).(*tmysql.MySQLError)
+	if !ok {
+		return -1, false
+	}
+
+	return terror.ErrCode(mysqlErr.Number), true
+}
+
+func isUnknownSystemVariableErr(err error) bool {
+	code, ok := getSQLErrCode(err)
+	if !ok {
+		return strings.Contains(err.Error(), "Unknown system variable")
+	}
+	// ErrUnknownSystemVariable
+	return code == 1193
```

Comment:
you could use the ErrUnknownSystemVariable constant from `github.com/pingcap/parser/mysql`...